### PR TITLE
feat(sourcemaps): simplified source map upload view 

### LINKED ISSUE #77400: Improve source maps view in releases
Currently, the [view for source maps](https://sentry-sdks.sentry.io/settings/projects/lforst/source-maps/artifact-bundles/) is barely usable for all common use cases.

Generally speaking this view should support the following use cases:
- I want to delete all files from a particular upload.
- I want to see all files for a particular upload.
- I want to globally search for a file by debug ID.
- Within a particular upload, I want to search for a file by name.

Current issues I see:
- Searching globally for a debug ID filters artifact bundles.
- The release bundles tab is useless and features wrong information. (0 artifacts?)
- Uploads are split into artifact bundles seemingly arbitrarily. What is the heuristic?
- What are artifact bundles? What are release bundles? (This is not a documentation issue. These concepts simply shouldn't exist for the users.)
- Why does a bundle say "associated release**S**"? How is that even possible? Is it possible for an artifact bundle to have multiple releases?
- We show the user random IDs in a list.

### PR DESCRIPTION
**Pull request by obostjancic**
- closes #77400 
- Adds an UI that unifies Artifact and Release bundles 

Before:
image

After:
image

### CODE REVIEW

File: static/app/views/settings/projectSourceMaps/projectSourceMapsUploads.tsx [hunk 1]
```diff
@@ -0,0 +1,389 @@
...
+          `These source map archives help Sentry identify where to look when code is minified. By providing this information, you can get better context for your stack traces when debugging. To learn more about source maps, [link: read the docs].`,
+          {
+            link: (
+              <ExternalLink href="https://docs.sentry.io/platforms/javascript/sourcemaps/" />
+            ),
+          }
+        )}
+      </TextBlock>
+      <SearchBarWithMarginBottom
+        placeholder={t('TODO Filter by Debug ID or Upload ID')}
```
Comment (shellmayr) [inline static/app/views/settings/projectSourceMaps/projectSourceMapsUploads.tsx:168]: Remove TODO

**Review by shellmayr — APPROVED** [review]

File: static/app/views/settings/projectSourceMaps/projectSourceMapsUploads.tsx (hunk 1 above)
Comment (obostjancic) [reply static/app/views/settings/projectSourceMaps/projectSourceMapsUploads.tsx]: thx

File: static/app/views/settings/projectSourceMaps/index.tsx [hunk 2]
```diff
@@ -1,18 +1,76 @@
...
   {}
 > & {
   children: React.ReactNode;
   project: Project;
 };
 
+const hasSourceMapUploadsView = (org: Organization) =>
+  org.features.includes('new-source-map-uploads-view');
+
+export function ToggleSourceMapUploadsView({router}) {
```
Comment (priscilawebdev) [inline static/app/views/settings/projectSourceMaps/index.tsx:33]: shouldn't we add types here? 🙏

File: static/app/views/settings/projectSourceMaps/index.tsx [hunk 3]
```diff
@@ -1,18 +1,76 @@
...
+  const useSourceMapUploadsView =
+    router.location.query.useSourceMapUploadsView === 'true';
```
Comment (priscilawebdev) [inline static/app/views/settings/projectSourceMaps/index.tsx:35]:
```suggestion
  const hasSourceMapUploadsView =
    router.location.query.useSourceMapUploadsView === 'true';
```
it is just a suggestion as `use` is normally used to indicate a hook

File: static/app/views/settings/projectSourceMaps/index.tsx [hunk 4]
```diff
@@ -1,18 +1,76 @@
...
+
+  const org = useOrganization();
+
+  const toggle = useCallback(() => {
+    const newVal = useSourceMapUploadsView ? undefined : 'true';
+    router.push({
+      ...location,
+      query: {...router.location.query, useSourceMapUploadsView: newVal},
+    });
```
Comment (priscilawebdev) [inline static/app/views/settings/projectSourceMaps/index.tsx:44]: shouldn't we use `navigateTo` instead? I remember seeing a deprecated warning when using `router.push` a few days ago

File: static/app/views/settings/projectSourceMaps/index.tsx [hunk 5]
```diff
@@ -1,18 +1,76 @@
...
+  if (!hasSourceMapUploadsView(org)) {
+    return null;
+  }
+
+  return (
+    <SwitchButtonContainer
+      style={{
+        opacity: useSourceMapUploadsView ? 1.0 : 0.75,
+        gap: space(1),
+      }}
```
Comment (priscilawebdev) [inline static/app/views/settings/projectSourceMaps/index.tsx:56]: nit: I believe this code is just temporary but we avoid using `style`

File: static/app/views/settings/projectSourceMaps/index.tsx [hunk 6]
```diff
@@ -22,6 +80,23 @@ export default function ProjectSourceMapsContainer({params, location, ...props}:
       />
     );
   }
+  if (location.query.useSourceMapUploadsView === 'true' && hasSourceMapUploadsView(org)) {
```
Comment (priscilawebdev) [inline static/app/views/settings/projectSourceMaps/index.tsx:83]:
```suggestion
  if (useSourceMapUploadsView&& hasSourceMapUploadsView(org)) {
```

File: static/app/views/settings/projectSourceMaps/index.tsx [hunk 7]
```diff
@@ -22,6 +80,23 @@ export default function ProjectSourceMapsContainer({params, location, ...props}:
...
+    return (
+      <Fragment>
+        <ToggleSourceMapUploadsView router={props.router} />
+        <ProjectSourceMapsUploads
+          {...props}
+          location={location}
+          params={{...params, bundleId: params.bundleId!}}
```
Comment (priscilawebdev) [inline static/app/views/settings/projectSourceMaps/index.tsx:90]: can the `bundleId` be `undefined` here?

File: static/app/views/settings/projectSourceMaps/projectSourceMapsUploads.tsx [hunk 8]
```diff
@@ -0,0 +1,389 @@
...
+            {showAll ? t('Show Less') : t('Show All')}
+          </Button>
+        ),
+        value:
+          rows.length > 0 ? (
+            <ReleasesWrapper className="val-string-multiline">
+              {visibleAssociations.map(association => (
+                <Fragment key={association.release}>
+                  <Version version={association.release ?? association.dist} />
+                  {association.dist && `(Dist: ${formatDist(association.dist)})`}
```
Comment (priscilawebdev) [inline static/app/views/settings/projectSourceMaps/projectSourceMapsUploads.tsx:282]: shall we use a translation function here?

File: static/app/views/settings/projectSourceMaps/projectSourceMapsUploads.tsx [hunk 9]
```diff
@@ -0,0 +1,389 @@
...
+
+  return <StyledKeyValueList data={detailsData} shouldSort={false} />;
+}
+
+const formatDist = (dist: string | string[] | null) => {
+  if (Array.isArray(dist)) {
+    return dist.join(', ');
+  }
+  if (dist === null) {
+    return 'none';
```
Comment (priscilawebdev) [inline static/app/views/settings/projectSourceMaps/projectSourceMapsUploads.tsx:301]: shall we use a translation function here?

File: static/app/views/settings/projectSourceMaps/projectSourceMapsUploads.tsx [hunk 10]
```diff
@@ -0,0 +1,389 @@
...
+          </Confirm>
+        </Tooltip>
+      )}
+    </Access>
+  );
+}
+
+const ReleasesWrapper = styled('pre')`
+  max-height: 200px;
+  overflow-y: auto !important;
```
Comment (priscilawebdev) [inline static/app/views/settings/projectSourceMaps/projectSourceMapsUploads.tsx:348]: we can also use `&&` to override the style 😉

**Comment by priscilawebdev** [comment]
do we have unit tests?

**Pull request merged**